tasks/pdr-cleanup/test/test_pdr_cleanup.py

```diff
@@ -0,0 +1,138 @@
+import os
+from datetime import datetime
+from unittest.mock import MagicMock, patch
+
+import pytest
+from pdr_cleanup import cleanup_pdr, move_pdr
+
+
+@pytest.fixture
+def successful_event():
+    return {
+        "config": {
+            "provider": {
+                "host": "test-bucket",
+                "protocol": "s3",
+            },
+        },
+        "input": {
+            "pdr": {
+                "name": "test-pdr.PDR",
+                "path": "dropbox",
+            },
+            "failed": [],
+        },
+    }
+
+
+@pytest.fixture
+def failed_event():
+    return {
+        "config": {
+            "provider": {
+                "host": "test-bucket",
+                "protocol": "s3",
+            },
+        },
+        "input": {
+            "pdr": {
+                "name": "test-pdr.PDR",
+                "path": "dropbox",
+            },
+            "failed": [
+                {"arn": 123, "reason": "failure"},
+                {"arn": 456, "reason": "failure"},
+            ],
+        },
+    }
+
+
+@pytest.fixture
+def mock_context():
+    return {}
+
+
+@patch("pdr_cleanup.move_pdr")
+def test_cleanup_pdr_success(mock_move_pdr, successful_event, mock_context):
+    output = cleanup_pdr(successful_event, mock_context)
+
+    mock_move_pdr.assert_called_once_with(
+        successful_event["config"]["provider"], successful_event["input"]["pdr"]
+    )
+
+    assert output == successful_event["input"]
+
+
+def test_cleanup_pdr_with_failed_workflows(failed_event, mock_context):
+    with pytest.raises(Exception):
+        cleanup_pdr(failed_event, mock_context)
+
+
+@patch("pdr_cleanup.boto3.client")
+@patch("pdr_cleanup.datetime")
+def test_move_pdr_success(mock_datetime, mock_boto3_client, successful_event):
+    mock_datetime.now.return_value = datetime(2025, 4, 2, 1, 1)
+    mock_s3_client = MagicMock()
+    mock_boto3_client.return_value = mock_s3_client
+
+    provider = successful_event["config"]["provider"]
+    pdr = successful_event["input"]["pdr"]
+
+    src_path = os.path.join(pdr["path"], pdr["name"])
+    dest_path = os.path.join("PDRs", pdr["path"], "2025.04.02", pdr["name"])
+
+    move_pdr(provider, pdr)
+
+    mock_s3_client.copy_object.assert_called_once_with(
+        CopySource=os.path.join(provider["host"], src_path),
+        Bucket=provider["host"],
+        Key=dest_path,
+    )
```

Comment:
Although, to go further, `moto` is a great library for mocking AWS calls as it will essentially simulate the real AWS API, allowing you to get the parameter validation part of your `boto3` calls which can sometimes catch some otherwise missed bugs. 

I can't recommend `moto` highly enough. https://github.com/getmoto/moto